XmlMerger requires merge:id for single attribute

### Actual behavior

```
Caused by: java.lang.IllegalStateException: Attribute merge:id is required for XML element /code_scheme[@name='ide' @version='173']/codeStyleSettings[@language='Groovy']
        at com.devonfw.tools.ide.merge.xml.matcher.ElementMatcher.getIdComputer(ElementMatcher.java:43)
        at com.devonfw.tools.ide.merge.xml.matcher.ElementMatcher.matchElement(ElementMatcher.java:60)
        at com.devonfw.tools.ide.merge.xml.XmlMergeStrategy.combineChildNodes(XmlMergeStrategy.java:106)
        at com.devonfw.tools.ide.merge.xml.XmlMergeStrategy$1.doMerge(XmlMergeStrategy.java:29)
        at com.devonfw.tools.ide.merge.xml.XmlMergeStrategy.merge(XmlMergeStrategy.java:64)
        ... 28 more
```

For XML snippet:
```
<code_scheme name="Default" version="173" xmlns:merge="https://github.com/devonfw/IDEasy/merge">
  <!-- ... -->
  <codeStyleSettings language="Groovy">
    <indentOptions>
      <option name="INDENT_SIZE" value="2"/>
    </indentOptions>
  </codeStyleSettings>
  <!-- ... -->
</code_scheme>
```

### Reproduce

with current configuration simply do:
1. ide create test123 -
2. icd test123
3. ide android-studio
4. watch the logged exceptions

### Expected behavior

As documented [here](https://github.com/devonfw/IDEasy/blob/main/documentation/configurator.adoc#element-identification) the default `merge:id` for an element that only has one single attribute should be the value of that attribute.
> Otherwise, if only a single attribute is present, that one will be used.

### IDEasy status

```bash
$ ide -p status
Updates are available for the settings repository. If you want to apply the latest changes, call "ide update"
IDE_ROOT is set to D:/projects
IDE_HOME is set to D:/projects/project
You are online.
Your settings are not up-to-date, please run 'ide update'.
Your version of IDEasy is 2025.07.002.
Your are using the latest version of IDEasy and no update is available.
Your operating system is windows(10.0)@x64 [Windows 11@amd64]
```

### Related/Dependent issues

621

### Comments/Hints

The bug is trivial:
https://github.com/devonfw/IDEasy/blob/68b8a3b17124ba75bbcbe5105617c65d35649cc6/cli/src/main/java/com/devonfw/tools/ide/merge/xml/XmlMergeSupport.java#L287

Simply replace `!=` with `==` - what a classic.

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -7,6 +7,7 @@ This file documents all notable changes to https://github.com/devonfw/IDEasy[IDE
 Release with new features and bugfixes:
 
 * https://github.com/devonfw/IDEasy/issues/1443[#1443]: Tool version is being added to settings even though it does not exist
+* https://github.com/devonfw/IDEasy/issues/1451[#1451]: XmlMerger requires merge:id for single attribute
 
 The full list of changes for this release can be found in https://github.com/devonfw/IDEasy/milestone/33?closed=1[milestone 2025.08.001].
 

--- a/cli/src/main/java/com/devonfw/tools/ide/merge/xml/XmlMergeSupport.java
+++ b/cli/src/main/java/com/devonfw/tools/ide/merge/xml/XmlMergeSupport.java
@@ -284,7 +284,7 @@ public interface XmlMergeSupport {
     for (int i = 0; i < attributeCount; i++) {
       Attr attribute = (Attr) attributes.item(i);
       if (!hasMergeNamespace(attribute)) {
-        if (id != XPATH_ELEMENT_NAME) {
+        if (id == XPATH_ELEMENT_NAME) {
           id = "@" + attribute.getName();
         } else {
           id = null;

--- a/cli/src/test/resources/xmlmerger/id-fallback/result.xml
+++ b/cli/src/test/resources/xmlmerger/id-fallback/result.xml
@@ -10,5 +10,8 @@
   </parent>
   <special class="id2">new special2</special>
   <special class="id3">old special3</special>
+  <special2 class="id5" key="id6">new special4</special2>
+  <special2 class="id7">old special5</special2>
   <special class="id1">new special1</special>
+  <special2 class="id3" key="id4">new special3</special2>
 </root>

--- a/cli/src/test/resources/xmlmerger/id-fallback/target.xml
+++ b/cli/src/test/resources/xmlmerger/id-fallback/target.xml
@@ -10,4 +10,6 @@
   </parent>
   <special class="id2">old special2</special>
   <special class="id3">old special3</special>
+  <special2 class="id5">old special4</special2>
+  <special2 class="id7">old special5</special2>
 </root>

--- a/cli/src/test/resources/xmlmerger/id-fallback/template.xml
+++ b/cli/src/test/resources/xmlmerger/id-fallback/template.xml
@@ -6,4 +6,6 @@
   </parent>
   <special class="id1">new special1</special>
   <special class="id2">new special2</special>
+  <special2 class="id3" key="id4" merge:id="@class">new special3</special2>
+  <special2 class="id5" key="id6">new special4</special2>
 </root>

--- a/cli/src/test/resources/xmlmerger/id-fallback/template.xml
+++ b/cli/src/test/resources/xmlmerger/id-fallback/template.xml
@@ -4,6 +4,6 @@
   <parent id="parent" attr1="newSomething">
     <child>New Child Text</child>
   </parent>
-  <special class="id1" merge:id="@class">new special1</special>
+  <special class="id1">new special1</special>
   <special class="id2">new special2</special>
 </root>

